kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/RecordEncryptionFilter.java

```diff
@@ -57,6 +61,27 @@ public class RecordEncryptionFilter<K>
         this.encryptionManager = encryptionManager;
         this.decryptionManager = decryptionManager;
         this.filterThreadExecutor = filterThreadExecutor;
+        checkCipherSuite();
+    }
+
+    private static void checkCipherSuite() {
+        checkCipherSuite(CipherSpec::newCipher);
+    }
+
+    /* exposed for testing */ static void checkCipherSuite(Function<CipherSpec, Cipher> cipherFunc) {
+        List<CipherSpec> failures = Arrays.stream(CipherSpec.values()).flatMap(cipherSpec -> {
```

Comment:
Probably worth commenting somewhere around here why we check **all** CipherSpecs rather than just the configured.

Reply:
Yeah, I wonder if we should have some configuration available l, maybe the user could supply an optional list of supported ciphers. So they can opt out of a ciphers altogether if their platform doesn't support it

Reply:
yeah thats a good point we need a filter too.